src/site/asciidoc/news.adoc

```diff
@@ -15,6 +15,67 @@
+
+==== New JSON parser
+
+Since https://github.com/jolokia/jolokia/commit/4d7a089e97576836945983ecfbbc92ae818b50ac[first available GitHub commit], Jolokia was using `com.googlecode.json-simple:json-simple` JSON parser/serializer library. This library has quite an interesting history (moving from https://code.google.com/archive/p/json-simple/[code.google.com] to GitHub, forking, ...). It does its job very well, but we've decided to make a bold move and remove this dependency without adding a new external JSON dependency.
+
+Here's what we did - we used https://github.com/fangyidong/json-simple/blob/tag_release_1_1_1/doc/json.lex[JSON grammar used by `json-simple`], checked the https://github.com/cliftonlabs/json-simple/blob/json-simple-4.0.1/src/main/lex/jsonstrict.lex[changes made in forked version] and used https://www.jflex.de/[JFlex] to generate Jolokia own lexer. +
+With a working lexer, we've built our flexible parser/serializer to fulfill Jolokia's needs. And yes, we've made it ~2x faster (in non-exhaustive tests involving https://activemq.apache.org/components/artemis/[Artemis broker] running with 10,000 queues).
+
+The new Jolokia module is `org.jolokia:jolokia-json` and it fully replaces `com.googlecode.json-simple:json-simple` library. Class names are the same, the only change is in the package name (switch from `org.json.simple` to `org.jolokia.json`).
+
+==== New Jolokia protocol version
+
+Going with the flow, after moving from JavaScript to TypeScript and getting our own JSON parser, we've introduced a new Jolokia protocol version, `8.0`. There's more information in the link:reference/html/manual/jolokia_protocol.html#optimized-response-list[relevant documentation section], but here's a summary of the changes:
+
+* With the new `listKeys` processing/request parameter, `list()` operation returns an additional `"keys"` field for MBean's MBeanInfo - it contains a map of the keys obtained from ObjectName
+* With the new `includeRequest` processing/request parameter, we can tell Jolokia not to return incoming requests under `the "request"` field of the response (defaults to `false`)
+* The' list () ' operation may be optimized with the new `listCache` processing/request parameter. It allows caching the common JSON-ified MBeanInfo to avoid duplicating it in the `list()` response for, say, 50,000 Artemis queue MBeans. Getting 2MB instead of 0.5GB of JSON is always a good thing™. More details can be found in the documentation.
+
+==== New extension points
+
+To make `list()` operation more flexible (based on experience gained from the https://hawt.io/[Hawtio console])), Jolokia now provides extension points for `list()`. While it is still based on `javax.management.MBeanServerConnection.queryMBeans()` call, we don't necessarily blindly serialize each MBean's `javax.management.MBeanInfo`.
```

Comment:
(based on experience gained from the https://hawt.io/[Hawtio console])) --> (based on experience gained from thehttps://hawt.io/[Hawtio console])

Reply:
ah yes - this is important: https://xkcd.com/859/